### PR TITLE
fix: force raw codec if requested

### PR DESCRIPTION
**Pull request by Arqu**
Closes https://github.com/n0-computer/iroh/issues/456

### CODE REVIEW

File: iroh-gateway/src/core.rs [hunk 1]
```diff
@@ -396,6 +396,42 @@ mod tests {
...
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            &format!("/ipfs/{}/hello.txt?format=raw", test_setup.root_cid),
+            None,
+        )
+        .await;
+        assert_eq!(http::StatusCode::OK, res.status());
+        let body = hyper::body::to_bytes(res.into_body()).await.unwrap();
+        let ufs = iroh_unixfs::unixfs::UnixfsNode::decode(&test_setup.root_cid, body).unwrap();
```
Comment (Arqu) [inline iroh-gateway/src/core.rs:431]: Am slightly confused on how to unpack this further without significantly complicating things.

File: iroh-resolver/src/resolver.rs [hunk 2]
```diff
@@ -781,18 +781,35 @@ impl<T: ContentLoader> Resolver<T> {
...
         // Resolve the root block.
         let (root_cid, loaded_cid) = self.resolve_root(&path, &mut ctx).await?;
         match loaded_cid.source {
             Source::Store(_) => inc!(ResolverMetrics::CacheHit),
             _ => inc!(ResolverMetrics::CacheMiss),
         }
 
-        let codec = Codec::try_from(root_cid.codec()).context("unknown codec")?;
+        let codec = match force_raw {
+            true => Codec::Raw,
```
Comment (dignifiedquire) [inline iroh-resolver/src/resolver.rs:810]: this works, but is really misusing the current setup

Comment (dignifiedquire) [reply iroh-resolver/src/resolver.rs:810]: but I guess it works for now

**Comment by Arqu** [comment]
@dignifiedquire where we at with this?

**Review by dignifiedquire — APPROVED** [review]
lets ship it for now

**Pull request merged**